### ISSUE TITLE
dive deeper into valid/invalid inputs for forms

### ISSUE THREAD

**Issue by matthewcarleton** [patternfly/patternfly-next#587]
This is related to #504 

Based on @jgiardino's review I think we need to break out some of the accessibility work to test things further. 

This will require further discussion but it revolves around more extensive testing and what that looks like in the workspace.

**Comment by matthewcarleton** [comment]
@kybaker @mceledonia through working on this I've come across a few things we should address.

when on a dark background the error message for inputs doesn't have proper contrast. Can we come up with a solution for that?

Also, for the readonly input the background colour used is pf-color-black-150 we don't use this colour currently as a background colour. Can we revisit this in design rather than introduce that colour just for this use?

**Comment by mceledonia** [comment]
@matthewcarleton Thanks for finding those, we definitely need to address them.

For read only, we should be fine bumping the background color up to pf-black-200, keeping the outline color as is.

For the input error on dark background, the first thing I would try is using the lighter red (CC0000) when on dark background with the theming system if possible. If that doesn't pass we may need to revisit how we visualize input errors and try something with a background color... Thoughts @kybaker ?

**Comment by kybaker** [comment]
@mceledonia @matthewcarleton We might be able to take advantage of the CSS color theming so we can come up with an alternate red. I would like to keep the red you had defined over the white if possible. We want to avoid cc000. Another idea is to use an outerglow like we had in PF3. That would change the contrast requirements.

**Comment by matthewcarleton** [comment]
@kybaker @mceledonia I've tried #cc0000 and no luck there. We can explore this further when I swing back around to this later this week.

**Comment by matthewcarleton** [comment]
@mattnolting implementing your above suggestion. Question about the aria-invalid conditional. Do we need an `else` there if the default is false?
